File: Assignment14/Q2.c
///////////////////////////////////////////////////////////////////
//
//      Including Header Files
//
///////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>

///////////////////////////////////////////////////////////////////
//
//      Function name:  Frequency
//      Input:          Integer Array, Integer
//      Output:         Integer
//      Description:    Accept array and return difference between
//                      frequency of even numbers and odd numbers
//      Date:           03/06/2026
//      Author:         Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

int Frequency(
                int Arr[],     // Input array
                int iLength    // Number of elements
             )
{
    int iCnt = 0;              // Loop counter
    int iEvenCnt = 0;          // Stores frequency of even numbers
    int iOddCnt = 0;           // Stores frequency of odd numbers

    // Traverse the array
    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        // Check whether element is even
        if((Arr[iCnt] % 2) == 0)
        {
            iEvenCnt++;
        }
        else
        {
            iOddCnt++;
        }
    }

    return (iEvenCnt - iOddCnt);
}

///////////////////////////////////////////////////////////////////
//
//      Entry point function
//
///////////////////////////////////////////////////////////////////

int main()
{
    int iSize = 0;             // Stores number of elements
    int iRet = 0;              // Stores return value from function
    int iCnt = 0;              // Loop counter

    int *p = NULL;             // Pointer for dynamic memory allocation

    printf("Enter number of elements\n");
    scanf("%d",&iSize);

    // Allocate memory dynamically
    p = (int *)malloc(iSize * sizeof(int));

    // Check whether memory allocation is successful
    if(p == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    printf("Enter %d elements\n",iSize);

    // Accept elements from user
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element : %d\n",iCnt + 1);
        scanf("%d",&p[iCnt]);
    }

    // Function call
    iRet = Frequency(p, iSize);

    // Display result
    printf("%d\n",iRet);

    // Deallocate memory
    free(p);

    return 0;
}

///////////////////////////////////////////////////////////////////
//
//      Input :
//      N : 7
//
//      Elements :
//      85 66 3 80 93 88 90
//
//      Output :
//      1      (4 - 3)
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
//      Where n is the number of elements in array
//
///////////////////////////////////////////////////////////////////

File: Assignment21/Q2.java
///////////////////////////////////////////////////////////////////
//
//      File Name : Q2.java
//      Project   : Logic Building Assignment 21
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : EvenOddRangeLogic
//      Method Name: countEvenOddRange
//      Input      : Integer
//      Output     : Void
//      Description: Count total even and odd
//                   numbers between 1 and N
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class EvenOddRangeLogic
{
    void countEvenOddRange(int n)
    {
        int i = 0;          // Loop counter
        int iEven = 0;      // Store count of even numbers
        int iOdd = 0;       // Store count of odd numbers

        // Count even and odd numbers
        for(i = 1; i <= n; i++)
        {
            if(i % 2 == 0)
            {
                iEven++;
            }
            else
            {
                iOdd++;
            }
        }

        // Display result
        System.out.println("Even Numbers : " + iEven);
        System.out.println("Odd Numbers  : " + iOdd);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q2
{
    public static void main(String args[])
    {
        // Create object of EvenOddRangeLogic class
        EvenOddRangeLogic obj = new EvenOddRangeLogic();

        // Function call
        obj.countEvenOddRange(50);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 50
//
//      Output :
//      Even Numbers : 25
//      Odd Numbers  : 25
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(n)
//
//      Space Complexity : O(1)
//
///////////////////////////////////////////////////////////////////
